dapps/digital-twin-detail/src/components/sharing/SharingOverview.ts

```diff
@@ -114,23 +131,24 @@ export default class SharingOverview extends mixins(EvanComponent) {
     const runtime = this.getRuntime();
     const { containerAddresses } = this.$store.state.twin;
     const permittedUsers: { [address: string]: LoadingSharedUserInterface } = { };
-
+    // reset previous data
+    this.data = [];
     // iterate through all containers and resolve the permissions
     await Promise.all(containerAddresses.map(async (containerAddress: string): Promise<void> => {
       /* Hint: Container API invites every shared user generally into the member role. As a result
        * of this, we can simply resolve all members from the rights and roles, to check with which
        * users anything was shared. */
       const contract = runtime.contractLoader.loadContract('DataContract', containerAddress);
       const roleMap = await runtime.rightsAndRoles.getMembers(contract);
-      const unique = new Set(Object.values(roleMap));
+      const unique = new Set([].concat(...Object.values(roleMap)));
```

Comment:
why `[].concat` ?

Reply:
It's like the following sample:

![image](https://user-images.githubusercontent.com/7114819/74413206-483bb480-4e3f-11ea-91eb-baebede465f3.png)

If we are not using concat, it will only apply the first role to the set of addresses.

Reply:
This would do the trick aswell, while being more declarative: